git-hooks/install.sh

```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
```

Comment:
1. What happens if git-secrets is not installed? Presumably/hopefully this script fails and in turn causes the build to fail?
2. Do you need the oft recommended by Denis:
```
set -o errexit
set -o nounset
set -o pipefail
```
3. Every time a build is done, this will remove hooks, call `git secrets --register-aws`, and add hooks again. Is that OK? If the `git secrets --register-aws` is idempotent, should be fine, but just checking.

Reply:
On further reflection -- you can't have it fail without checking Travis and Circle env variables, or it will fail there.

Overall, this PR is a big improvement, but if somebody doesn't have git secrets installed (new machine, new engineer), things could still slip through. It would be nice if commits would actually fail for all non-CI users.

Reply:
> What happens if git-secrets is not installed? Presumably/hopefully this script fails and in turn causes the build to fail?

You can test this by moving your git secrets binary, the build error looks like this:

```
[INFO] Reactor Summary for dockstore 1.10.0-alpha.1-SNAPSHOT:
[INFO] 
[INFO] dockstore .......................................... SUCCESS [  0.485 s]
[INFO] dockstore-common ................................... SUCCESS [  2.033 s]
[INFO] dockstore-language-plugin-parent ................... SUCCESS [  0.648 s]
[INFO] swagger-java-quay-client ........................... SUCCESS [  0.118 s]
[INFO] swagger-java-sam-client ............................ SUCCESS [  0.047 s]
[INFO] swagger-java-bitbucket-client ...................... SUCCESS [  0.048 s]
[INFO] swagger-java-discourse-client ...................... SUCCESS [  0.045 s]
[INFO] swagger-java-zenodo-client ......................... SUCCESS [  0.077 s]
[INFO] dockstore-webservice ............................... FAILURE [  2.343 s]
[INFO] swagger-java-client ................................ SKIPPED
[INFO] openapi-java-client ................................ SKIPPED
[INFO] dockstore-integration-testing ...................... SKIPPED
[INFO] dockstore-event-consumer ........................... SKIPPED
[INFO] reports ............................................ SKIPPED
[INFO] bom-internal ....................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  6.069 s
[INFO] Finished at: 2020-09-04T11:58:49-07:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.6.0:exec (Install git-secrets) on project dockstore-webservice: Command execution failed.: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :dockstore-webservice
```

A developer technically can still download source, make changes and not perform the build. You will need to have git-secrets installed in environments where the build is happening, so this will need to be added to CI to merge.

https://travis-ci.org/github/dockstore/dockstore/jobs/724273630#L3707

> Do you need the oft recommended by Denis (mode set)

Will add

> Every time a build is done, this will remove hooks, call git secrets --register-aws, and add hooks again. Is that OK? If the git secrets --register-aws is idempotent, should be fine, but just checking.

I did verify the register call is idempotent, you can see by looking out the output of `git config -l`, which shows you all the config keys set for the current git directory. The `git config --add` call, however, is not idempotent, hence some logic was added to unset the config before setting it again

If we want to set other git secrets patterns that are portable to other developers, this is an OK place to do that.